x-pack/plugins/security_solution/server/lib/detection_engine/signals/mappings/build_severity_from_mapping.ts

```diff
@@ -31,41 +32,89 @@ const severitySortMapping = {
   critical: 3,
 };
 
+const ECS_SEVERITY_FIELD = 'event.severity';
+
 export const buildSeverityFromMapping = ({
   eventSource,
   severity,
   severityMapping,
 }: BuildSeverityFromMappingProps): BuildSeverityFromMappingReturn => {
-  if (severityMapping != null && severityMapping.length > 0) {
-    let severityMatch: SeverityMappingItem | undefined;
-
-    // Sort the SeverityMapping from low to high, so last match (highest severity) is used
-    const severityMappingSorted = severityMapping.sort(
-      (a, b) => severitySortMapping[a.severity] - severitySortMapping[b.severity]
-    );
-
-    severityMappingSorted.forEach((mapping) => {
-      const docValue = get(mapping.field, eventSource);
-      // TODO: Expand by verifying fieldType from index via doc._index
-      // Till then, explicit parsing of event.severity (long) to number. If not ECS, this could be
-      // another datatype, but until we can lookup datatype we must assume number for the Elastic
-      // Endpoint Security rule to function correctly
-      let parsedMappingValue: string | number = mapping.value;
-      if (mapping.field === 'event.severity') {
-        parsedMappingValue = Math.floor(Number(parsedMappingValue));
-      }
-
-      if (parsedMappingValue === docValue) {
-        severityMatch = { ...mapping };
-      }
-    });
-
-    if (severityMatch != null && SeverityIOTS.is(severityMatch.severity)) {
-      return {
-        severity: severityMatch.severity,
-        severityMeta: { severityOverrideField: severityMatch.field },
-      };
+  if (!severityMapping || !severityMapping.length) {
+    return defaultSeverity(severity);
+  }
+
+  let severityMatch: SeverityMappingItem | undefined;
+
+  // Sort the SeverityMapping from low to high, so last match (highest severity) is used
+  const severityMappingSorted = severityMapping.sort(
+    (a, b) => severitySortMapping[a.severity] - severitySortMapping[b.severity]
+  );
+
+  severityMappingSorted.forEach((mapping) => {
+    const eventField = mapping.field;
```

Comment:
Would it be more consistent to name this `mappingField` ?

Reply:
Will fix